### PR TITLE
Fix phase offset of Linkwitz-Riley Allpass Filters and add testing function

### PR DESCRIPTION
**Pull request by fietew**
For some filter orders (even, but not divisible by four), there was a wrong sign for LR Allpass Filters. This PR fix this and adds a testing function.

### CODE REVIEW

**Comment by fs446** [comment]
bug fix seems ok to me, just corrected the labeling of axis in the test script.

**Comment by hagenw** [comment]
The test runs fine o my machine.
Is this the desired output:
**Fig1**
fig1
**Fig 2**
fig2
**Fig 3**
fig3

File: validation/test_linkwitz_riley.m [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+%       modus   - 0: numerical
+%                 1: visual
+%
+%   Output parameters:
+%       status  - true or false
+
+%*****************************************************************************
+% The MIT License (MIT)                                                      *
+%                                                                            *
+% Copyright (c) 2010-2018 SFS Toolbox Developers                             *
```
Comment (hagenw) [inline validation/test_linkwitz_riley.m]: Please change `2018` to `2019` as I changed that already in the master.

File: validation/test_linkwitz_riley.m [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+status = false;
+
+%% ===== Checking of input  parameters ===================================
+nargmin = 1;
+nargmax = 1;
+narginchk(nargmin,nargmax);
+
+%% ===== Main ============================================================
+fs = 44100;  % sampling frequency
+fc = 1000;  % crossover frequency
```
Comment (hagenw) [inline validation/test_linkwitz_riley.m]: For both please add ` in Hz` at the end of the comment.

File: validation/test_linkwitz_riley.m [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  f, unwrap(angle(H(1,:))), 'b', ...
+  f, unwrap(angle(H(2,:))), 'r', ...
+  f, unwrap(angle(H(1,:)+H(2,:))), 'g.', ...
+  f, unwrap(angle(H(3,:))), 'k--' ...
+  );
+legend('Lowpass', 'Highpass', 'Lowpass + Highpass', 'Allpass', 'Location', 'southwest');
+xlabel('f / Hz');
+ylabel('angle(H) / rad')
+xlim([f(1), f(end)]);
+end
```
Comment (hagenw) [inline validation/test_linkwitz_riley.m]: Please add a newline at the end.

**Review by hagenw — CHANGES_REQUESTED** [review]

**Comment by fietew** [comment]
> Is this the desired output:

It seems, there is a problem regarding the Octave output. The greens lines in Fig1 and Fig3 (upper left) should be horizontal. I will have a look.

**Comment by fietew** [comment]
The behaviour is, again, related to a bug in zp2sos in Octave (see https://savannah.gnu.org/bugs/?51936), where single or non-existing zeros are handled incorrect. This was already mentioned in #169 . I used a similar workaround for this.

**Comment by hagenw** [comment]
Looks good now, here are the new figures I'm getting by running the test:
**Fig 1**
fig1
**Fig 2**
fig2
**Fig 3**
fig3

**Review by hagenw — APPROVED** [review]

**Review by fs446 — APPROVED** [review]
looks good

**Comment by hagenw** [comment]
Is it ok to squash the commits during merge?

Alternatively I would propose to condense them to two, one for the fix and one for adding testing functions.

**Pull request merged**